### PR TITLE
Make C.I work with packages installed through vcpkg

### PR DESCRIPTION
**Pull request by theophile-jr**
Featured a build.sh, CI running successfully, tests are launched, VCKPG is cached. 
Everything is ready to build good tests, running in continuous integration. 
The package manager setup and scripts have been improved on this PR alongside the issue.
image

image

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Cache vcpkg
+        uses: actions/cache@v4
+        with:
+          path: vcpkg
+          key: vcpkg-cache-${{ runner.os }}
+
+      - name: Apt install for Ubuntu
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          sudo apt install libxinerama-dev libxcursor-dev xorg-dev libglu1-mesa-dev pkg-config
```
Comment (mathieurobert1) [inline .github/workflows/checks.yml:25]: Why this for Ubuntu and not for Windows ?

File: Client/src/main.cpp [hunk 2]
```diff
@@ -6,13 +6,15 @@
 */
 
 #include <iostream>
+#include "asio.hpp"
```
Comment (mathieurobert1) [inline Client/src/main.cpp:9]: Remove this useless include.

File: Client/src/main.cpp [hunk 3]
```diff
@@ -6,13 +6,15 @@
...
 
 #include "Application.hpp"
 
 int main()
 {
     Application app;
-
+    
+    asio::io_context io;
```
Comment (mathieurobert1) [inline Client/src/main.cpp:17]: Remove this useless line

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]

File: .github/workflows/checks.yml (hunk 1 above)
Comment (theophile-jr) [reply .github/workflows/checks.yml:25]: Because it is required by cmake on ubuntu

File: Client/src/main.cpp (hunk 3 above)
Comment (theophile-jr) [reply Client/src/main.cpp:17]: Not mandatory but removing it. ;)

File: Client/src/main.cpp (hunk 2 above)
Comment (theophile-jr) [reply Client/src/main.cpp:9]: Not mandatory but removing it. ;)

File: .github/workflows/checks.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          if [ ! -d "vcpkg" ]; then
+            git clone https://github.com/microsoft/vcpkg.git
+            ./vcpkg/bootstrap-vcpkg.sh  
+          fi
+          ./vcpkg/vcpkg install
+
+      - name: Build project (Windows or Ubuntu)
+        run: |
+          cmake -B build -DCMAKE_TOOLCHAIN_FILE=${{ github.workspace }}/vcpkg/scripts/buildsystems/vcpkg.cmake
+          cmake --build build/ --config Debug
```
Comment (Thomaltarix) [inline .github/workflows/checks.yml:49]: Is the Debug specifier necessary ?

File: .github/workflows/checks.yml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+      - name: Run tests
+        run: |
+          cd build/tests/
+          ctest -C Debug
```
Comment (Thomaltarix) [inline .github/workflows/checks.yml:54]: Is the Debug specifier necessary ?

File: CMakeLists.txt [hunk 6]
```diff
@@ -20,3 +21,5 @@ add_subdirectory(Client)
 if(BUILD_TESTS)
     add_subdirectory(tests)
 endif()
+
+
```
Comment (Thomaltarix) [inline CMakeLists.txt:25]: Useless line
```suggestion

```

File: tests/Client/tests.cpp [hunk 7]
```diff
@@ -1,8 +1,17 @@
...
-{
-    cr_assert_eq("1", "1");
+void test() {
+    assert(1 == 1);
+}
+
+int main() {
+    test();
+    return 0;
 }
```
Comment (Thomaltarix) [inline tests/Client/tests.cpp:17]: Why is there a main here ?

File: tests/Ecs/tests.cpp [hunk 8]
```diff
@@ -1,8 +1,17 @@
...
-{
-    cr_assert_eq("1", "1");
+void test() {
+    assert(1 == 1);
+}
+
+int main() {
+    test();
+    return 0;
 }
```
Comment (Thomaltarix) [inline tests/Ecs/tests.cpp:17]: Same here, why is there a main ?

File: tests/Server/tests.cpp [hunk 9]
```diff
@@ -1,8 +1,18 @@
...
+#include <cassert>
+
+void test() {
+    assert(1 == 1);
+}
+
+int main() {
+    test();
+    return 0;
 }
```
Comment (Thomaltarix) [inline tests/Server/tests.cpp:18]: Same here, why is there a main ?

**Review by Thomaltarix — COMMENTED** [review]
Some questions more than requests, overall seems great 👍

**Review by mathieurobert1 — DISMISSED** [review]

**Review by Thomaltarix — APPROVED** [review]

**Review by mathieurobert1 — APPROVED** [review]

**Review by AxelF44 — APPROVED** [review]

File: build.sh
Comment (Peralban) [inline build.sh:1]: no blank line

**Review by Peralban — CHANGES_REQUESTED** [review]
It is possible to add system in cmake to check if we have the dependencies and install them.

**Comment by theophile-jr** [comment]
> It is possible to add system in cmake to check if we have the dependencies and install them.

Already done

**Review by Peralban — APPROVED** [review]

**Pull request merged**